spec/support/controller_helpers.rb

```diff
@@ -11,4 +11,18 @@ def current_user
   def user_signed_in?
     !current_user.nil?
   end
+
+  def mock_user(role = :user, **attrs)
+    spy('user', **FactoryGirl.attributes_for(role), **attrs)
+  end
+
+  def mock_user_sign_in(user = mock_user)
+    allow(request.env['warden']).to receive(:authenticate!).and_return(user)
+    # necessary because otherwise ApplicationController#app_setup_check fails
+    allow(User).to receive(:count).and_return(1)
```

Comment:
Shouldn't this live somewhere else then? It's not specifically necessary for sign in, but rather for any request to work.